src/search.cpp

```diff
@@ -161,8 +161,11 @@ void GreedySearch_Cpu::SampleTopK(int k, float temperature) {
     std::vector<int> indices(scores.size());
     std::iota(indices.begin(), indices.end(), 0);
     std::partial_sort(indices.begin(), indices.begin() + k, indices.end(), [scores = scores.data()](int i, int j) { return scores[i] > scores[j]; });
+    std::vector<float> top_k_scores(k);
+    for (int i = 0; i < k; i++)
+      top_k_scores.push_back(scores[indices[i]]);
```

Comment:
Here you're pre-allocating a vector of size k (and setting values to 0) and then pushing k more elements to it.

```suggestion
    std::vector<float> top_k_scores(k);
    for (int i = 0; i < k; i++)
      top_k_scores[i] = scores[indices[i]];
```

Reply:
Good catch, it still works since a bunch of 0's have no effect on the discrete_distribution. But not what is expected.